tests/methods/test_lime_timeseries.py

```diff
@@ -26,30 +26,55 @@ def test_lime_timeseries_correct_output_shape(self):
         )
         assert len(exp.local_exp[1]) == num_features
 
-    def test_cosine_distance(self):
-        """Test cosine distance."""
+    def test_distance_shape(self):
+        """Test the shape of returned distance array."""
         dummy_timeseries = np.random.random((50, 1))
         number_of_masks = 50
         masks = generate_masks(dummy_timeseries, number_of_masks, p_keep=0.9)
         masked = mask_data(dummy_timeseries, masks, mask_type="mean")
         explainer = LIMETimeseries()
         distance = explainer._calculate_distance(
-            dummy_timeseries, masked.reshape((-1, 50)), distance_method="cosine"
+            masked.reshape((-1, 50)), distance_method="cosine"
         )
         assert len(distance) == number_of_masks
 
+    def test_cosine_euclidean_distance(self):
+        """Test the calculation of cosine and euclidean distance."""
+        # Create some test data
+        masked_data = np.array([[1, 2, 3, 4, 5], [0, 0, 3, 4, 5]])
+
+        # Calculate the expected results
+        expected_cosine_distance = np.array([0.0, 4.653741])
+        expected_euclidean_distance = np.array([0.0, 2.236068])
+
+        # Calculate the distance
+        explainer = LIMETimeseries()
+
+        distance_cosine = explainer._calculate_distance(
+            masked_data, distance_method="cosine"
+        )
+
+        distance_euclidean = explainer._calculate_distance(
+            masked_data, distance_method="euclidean"
+        )
+
+        # Check that the calculated and expected results are equal
+        np.testing.assert_array_almost_equal(
+            [expected_cosine_distance, expected_euclidean_distance],
+            [distance_cosine, distance_euclidean],
+        )
+
```

Comment:
These are essentially two unit tests in one. I would suggest to split this up into two tests, this makes it easier to see what is going on when a test fails.

```suggestion
    def test_euclidean_distance(self):
        """Test the calculation of cosine and euclidean distance."""
        masked_data = np.array([[1, 2, 3, 4, 5], [0, 0, 3, 4, 5]])
        expected_distance = np.array([0.0, 2.236068])

        explainer = LIMETimeseries()
        distance = explainer._calculate_distance(masked_data,
                                                 distance_method="euclidean")
        np.testing.assert_array_almost_equal(distance, expected_distance)

    def test_cosine_distance(self):
        """Test the calculation of cosine and euclidean distance."""
        masked_data = np.array([[1, 2, 3, 4, 5], [0, 0, 3, 4, 5]])
        expected_distance = np.array([0.0, 4.653741])

        explainer = LIMETimeseries()
        distance = explainer._calculate_distance(masked_data,
                                                 distance_method="cosine")
        np.testing.assert_array_almost_equal(distance, expected_distance)

```

Reply:
Thanks for your review @stefsmeets. That makes sense to me. I will split it into two tests.